src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -70,22 +84,20 @@ public void updateOrder(String orderId, CompositePurchaseOrder compPO) {
     getPurchaseOrderById(orderId, lang, httpClient, ctx, okapiHeaders, logger)
       .thenCompose(poFromStorage -> validatePoNumber(compPO, poFromStorage))
       .thenCompose(poFromStorage -> updatePoLines(poFromStorage, compPO).thenApply(v -> poFromStorage))
-      .thenAccept(poFromStorage -> {
-        CompletableFuture<Void> updatedPoFuture;
+      .thenCompose(poFromStorage -> {
         if (isTransitionToOpen(compPO, poFromStorage)) {
-          updatedPoFuture = openOrder(compPO);
+          return openOrder(compPO)
+            .thenCompose(v -> changePoLinesReceiptStatus(compPO))
```

Comment:
Since the order can be created with the status "Open", it is necessary to provide for the possibility of updating the receipt status in this case. So I think it makes sense to call this in the "openOrder" method.

Reply:
done